### PR TITLE
🐛 Updates for the controller manager to watch everything

### LINKED ISSUE #1990: bug: Memory usage and logging issue of the controller manager when 'watch all'
### Describe the bug

Two things were observed when I ran the controller manager on OpenShift without the `api-groups` flag.

1. The controller manager got repeatedly `OOMKilled`.
    ```
    Every 2.0s: oc -n wds2-system get deploy,po                       Juns-MBP.local: Sat Mar 29 12:57:24 2024
    
    NAME                                             READY   UP-TO-DATE   AVAILABLE   AGE
    deployment.apps/kubestellar-controller-manager   0/1     1            0           26h
    deployment.apps/transport-controller             1/1     1            1           26h
    
    NAME                                                 READY   STATUS      RESTARTS      AGE
    pod/kubestellar-controller-manager-76666db99-zbpnk   1/2     OOMKilled   3 (46s ago)   117s
    pod/transport-controller-656fcff866-26mgt            1/1     Running     0             26h
    
    ```
3. The controller manager's log is full of "Got object event" messages similar to
    ```
    I0322 19:40:19.780018       1 controller.go:446] "Got object event" logger="Binding" eventType="add" wasDeletedFinalStateUnknown=false obj="ReplicaSet.apps(openshift-cluster-storage-operator/csi-snapshot-controller-operator-5589977cb8)" resource="replicasets"
    ```
    which quickly flushes away any other information.

### Steps To Reproduce

Run the controller manager
1. on an OpenShift cluster, or on a vanilla Kubernetes cluster with hundreds of GVRs;
2. without the `api-groups` flag;
3. check the controller manager's pod and log.

### Expected Behavior

No OOMKilled and better managed logging.

### Additional Context

Without the `api-groups` flag, the binding controller watches everything in the cluster. That is, all objects in etcd are fetched to the controller's cache, and the controller gets informed about every single event of the every single object.

We may want to raise (blindly, though) the memory limit of the pod, and tune the logging verbosity.

### PR DESCRIPTION
**Pull request by waltforme**
## Summary
This PR increases the memory limit of the controller manager, and adjusts the verbosity levels for the logging messages from the event handlers.

Changes in this PR are built into quay.io/junatibm/controller-manager:1711835357. I tested the image, with the increased memory limit, in the same environment as #1990. The controller ran for minutes without `OOMKilled`:
```
👉 Downloads $ oc cluster-info
Kubernetes control plane is running at https://c115-e.us-south.containers.cloud.ibm.com:30829

To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
👉 Downloads $ oc -n wds2-system get deploy,po
NAME                                             READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/kubestellar-controller-manager   1/1     1            1           8d
deployment.apps/transport-controller             1/1     1            1           8d

NAME                                                 READY   STATUS    RESTARTS   AGE
pod/kubestellar-controller-manager-56ccb5d76-4l5fz   2/2     Running   0          6m40s
pod/transport-controller-656fcff866-26mgt            1/1     Running   0          8d
👉 Downloads $ oc -n wds2-system get po kubestellar-controller-manager-56ccb5d76-4l5fz -oyaml | yq .spec.containers[1]
args:
  - --health-probe-bind-address=:8081
  - --metrics-bind-address=127.0.0.1:8080
  - --leader-elect
  - --wds-name=wds2
  - -v=2
image: quay.io/junatibm/controller-manager:1711835357
imagePullPolicy: IfNotPresent
livenessProbe:
  failureThreshold: 3
  httpGet:
    path: /healthz
    port: 8081
    scheme: HTTP
  initialDelaySeconds: 15
  periodSeconds: 20
  successThreshold: 1
  timeoutSeconds: 1
name: manager
readinessProbe:
  failureThreshold: 3
  httpGet:
    path: /readyz
    port: 8081
    scheme: HTTP
  initialDelaySeconds: 5
  periodSeconds: 10
  successThreshold: 1
  timeoutSeconds: 1
resources:
  limits:
    cpu: 500m
    memory: 1Gi
  requests:
    cpu: 10m
    memory: 64Mi
securityContext:
  allowPrivilegeEscalation: false
  capabilities:
    drop:
      - ALL
  runAsUser: 1001110000
terminationMessagePath: /dev/termination-log
terminationMessagePolicy: File
volumeMounts:
  - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
    name: kube-api-access-5zch7
    readOnly: true
👉 Downloads $ 
```

Also trying to hit #2000 by this PR.

## Related issue(s)
Fixes #1990

### CODE REVIEW

File: pkg/binding/controller.go [hunk 1]
```diff
@@ -455,7 +455,7 @@ func (c *Controller) handleObject(obj any, resource string, eventType string) {
 		obj = typed.Obj
 		wasDeletedFinalStateUnknown = true
 	}
-	c.logger.Info("Got object event", "eventType", eventType,
+	c.logger.V(3).Info("Got object event", "eventType", eventType,
```
Comment (MikeSpreitzer) [inline pkg/binding/controller.go:458]: I would make these V levels even higher. At least 4, maybe 5.

Comment (waltforme) [reply pkg/binding/controller.go]: Changed to 4.

**Review by MikeSpreitzer — COMMENTED** [review]
/lgtm

**Pull request merged**